src/Lib/Items/CanvasImage.vala

```diff
@@ -131,10 +157,14 @@ public class Akira.Lib.Items.CanvasImage : Goo.CanvasImage, Akira.Lib.Items.Canv
         manager.get_pixbuf.begin (w, h, (obj, res) => {
             try {
                 var _pixbuf = manager.get_pixbuf.end (res);
+                // TODO: run is_height_small and is_width_small
+                var pixpuf_width = _pixbuf.get_width ();
+                var pixpuf_height = _pixbuf.get_height ();
+
                 pixbuf = _pixbuf;
                 if (update) {
-                    width = _pixbuf.get_width ();
-                    height = _pixbuf.get_height ();
+                    width = pixpuf_width;
+                    height = pixpuf_height;
                 }
```

Comment:
mhhh...what's the reason for this?

Reply:
TODO: to call `is_height_small` and `is_width_small` with the `pixpuf_width, pixpuf_height` parameters that's why the methods take parameters

Reply:
No need to do this.
It's generally bad practice to update the code "in preparation for something else".
You're also affecting performance here as you're calling pixbuf methods which are heavy before are even needed, if any.